library/data/registry-entry-attributes/src/main/java/org/quiltmc/qsl/registry/attribute/api/RegistryExtensions.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.registry.attribute.api;
+
+import org.quiltmc.qsl.registry.attribute.impl.DefaultAttributesBuilderImpl;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import java.util.function.Consumer;
+
+/**
+ * Extensions for working with {@link Registry}s.
+ */
+public final class RegistryExtensions {
+	/**
+	 * Registers an entry into a registry, also setting built-in attribute values for it.
+	 *
+	 * @param registry target registry
+	 * @param id entry identifier
+	 * @param entry entry
+	 * @param builderConsumer builder consumer
+	 * @param <R> type of the entries in the registry
+	 * @param <T> type of the entry we're currently registering (may be a subclass of {@code R})
+	 * @return the newly registered entry
+	 */
+	public static <R, T extends R> T registerWithAttributes(Registry<R> registry, Identifier id, T entry,
+												  Consumer<DefaultAttributesBuilder<R>> builderConsumer) {
+		Registry.register(registry, id, entry);
+		builderConsumer.accept(new DefaultAttributesBuilderImpl<>(registry, entry));
+		return entry;
+	}
```

Comment:
Not sure how useful this method is. I can see its use case in mods, but it kinda seems like the only way to add attributes to entries with default values.

Reply:
That's because it is, mostly to let the registering mod determine what its entry should and should not have.

Ideally, if you were using an attribute, you'd have code to handle an entry which doesn't have a value assigned to said attribute, whether that would be a default value or an extra calculation that you'd cache yourself.

I _could_ add a `setAttributes` method here that allows you to mess with the attributes of an already-registered entry, if people want something like that.